test/channel.js

```diff
@@ -202,8 +202,42 @@ test('event channel', assert => {
 
   assert.ok(unsubscribeErr, 'eventChannel should throw if subscriber does not return a function to unsubscribe')
 
+  let unsubscribeWasCalled = false;
+  let chan = eventChannel(() => () => {
+    unsubscribeWasCalled = true;
+  });
+  chan.close();
+  assert.equal(unsubscribeWasCalled, true, 'eventChannel should call unsubscribe when channel is closed')
+
+  unsubscribeWasCalled = false;
+  chan = eventChannel((emitter) => {
+    emitter(END);
+    return () => {
+      unsubscribeWasCalled = true;
+    }
+  });
+  assert.equal(unsubscribeWasCalled, true, 'eventChannel should call unsubscribe when END event is emitted');
+
+  unsubscribeWasCalled = false;
+  let milliseconds = 2;
+  chan = eventChannel((emitter) => {
+    const interval = setInterval(() => {
```

Comment:
this could be rather simplified, all of this logic (complexity) boils down to the fact that `END` is emitted asynchronously, so simple `setTimeout(() => emitter(END), 0)` should be enough

+ ofc propert `assert.end()` should accompany this